packages/material-ui/src/Chip/Chip.js

```diff
@@ -322,63 +321,67 @@ class Chip extends React.Component {
           color !== 'default' && variant === 'outlined',
       };
 
-      deleteIcon =
-        deleteIconProp && React.isValidElement(deleteIconProp) ? (
-          React.cloneElement(deleteIconProp, {
-            className: clsx(deleteIconProp.props.className, classes.deleteIcon, customClasses),
-            onClick: this.handleDeleteIconClick,
-          })
-        ) : (
-          <CancelIcon
-            className={clsx(classes.deleteIcon, customClasses)}
-            onClick={this.handleDeleteIconClick}
-          />
-        );
+      return deleteIconProp && React.isValidElement(deleteIconProp) ? (
+        React.cloneElement(deleteIconProp, {
+          className: clsx(deleteIconProp.props.className, classes.deleteIcon, customClasses),
+          onClick: handleDeleteIconClick,
+        })
+      ) : (
+        <CancelIcon
+          className={clsx(classes.deleteIcon, customClasses)}
+          onClick={handleDeleteIconClick}
+        />
+      );
     }
+    return null;
+  }, [classes, color, deleteIconProp, handleDeleteIconClick, onDelete, variant]);
```

Comment:
`deleteIconProp` is an element, it will almost always invalidate the memoization. I would remove it.